Core/include/Acts/Propagator/CovarianceTransport.hpp

```diff
@@ -0,0 +1,113 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/TrackParametrization.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+
+#include <optional>
+#include <variant>
+
+namespace Acts {
+
+using VariantCovariance = std::variant<BoundSymMatrix, FreeSymMatrix>;
+
+using VariantTransportJacobian =
+    std::variant<BoundMatrix, BoundToFreeMatrix, FreeToBoundMatrix, FreeMatrix>;
+
+/// Helper struct holding the necessary cache for the Covariance
+/// transport in various parametrisations.
+struct CovarianceCache {
+  /// Internal cache state, indicates correct setup
+  bool applyTransport = false;
+
+  /// Internal cache state, at surface indication
+  const Surface* atSurface = nullptr;
```

Comment:
It's preparing for caching, imagine a second actor asks for the exact same covariance push to surface, I can catch it here.

Reply:
I tried to write down my imaginations but came back to the beginning: what do you mean with "covariance push to surface"? This surface in the context of covariance transport is used for the (re-)initialisation of the jacobians. After the covariance transport `atSurface` is the same as `state.navigation.currentSurface`.
I just cannot imagine the scenario :/

Reply:
Imagine you have an `Actor` for material integration and another one for hit finding, both would need the covariance matrix on the surface, the first one to add some noise, the second one to check compatibility with a Chi2 parameter. The second actor would call boundState() and it should not make another transport because the first actor had done this job already.

Reply:
Okay, thanks for the explanation and yes, this concept is reasonable. Wouldn't a bool store the same information then? Something that states "stepped since last covariance transport".
Since I cannot come up with a corresponding implementation right now that doesn't involve the setting of such a variable per step, I think your solution is fine